pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -141,6 +142,19 @@ def setup_environment(self) -> None:
 
         self.setup_distributed()
 
+        # share ddp pids to all processes
+        self._share_pids()
+
+    def _share_pids(self):
+        """
+        Make all DDP processes aware of all processes pids.
+        """
+        self.barrier()
+        pids = self.all_gather(torch.tensor(os.getpid(), device=self.root_device))
+        pids = ','.join(str(pid) for pid in pids.cpu().numpy().tolist())
+        os.environ["PL_INTERACTIVE_DDP_PROCS"] = pids
```

Comment:
They aren't interactive (like in jupyter), right?

also can we maybe store them somewhere in the plugin instead of the env variables? IMO we should only expose those environment variables if necessary.

Reply:
Done ! Yes, I am good with that.